teps/0084-endtoend-provenance-collection.md

```diff
@@ -153,14 +163,38 @@ In this proposal, we are suggesting following extensions for an end-to-end and c
+* **predicate.buildType**: Chains sets this value to `https://tekton.dev/attestations/chains@v2` for `taskruns` attestations. In order to facilitate consumption, a different value should be used: `https://tekton.dev/attestations/chains/pipelinerun@v2`.
+* **predicate.invocation**: For `taskruns`, Chains populates this object with the task parameters. Similarly, Chains must populate this object with `pipelinerun` parameters. Default parameters must also be added.
+* **predicate.buildConfig**: For `taskruns`, Chains creates an object with a single attribute, `steps`, which corresponds to all the steps within the `taskrun`. For `pipelineruns`, Chains must instead set the attribute `tasks` which is an array of objects representing each `taskrun` in the `pipelinerun`. Notice that skipped tasks are not represented in the attestation since they do not contribute to builiding the image. Each object contains following attributes:
+  * **name**: The name of the `task` (not the `taskrun`) within the `pipelinerun`.
+  * **after**: The name of the `task` (not the `taskrun`) within the `pipelinerun` that was configured to be executed prior to this `taskrun`. If none, this field is omitted. The list of tasks in `predicate.buildConfig.tasks` is sorted by execution time. However, `taskruns` within a `pipelinerun` can be executed in parallel. This field helps reconstruct the multi-dimensional execution order of `taskruns`.
+  * **status**: Either "Succeeded" or "Failed".
+  * **ref.kind**: Either `ClusterTask` or `Task`.
+  * **ref.name**: The name of the `task` (not the `taskrun`!) within the k8s cluster or within the Tekton Bundle.
+  * **ref.bundle**: Reference to the Tekton Bundle defining the task, or omitted if a Tekton Bundle was not used.
+  * **startedOn**: When the `taskrun` started.
+  * **finishedOn**: When the `taskrun` finished.
+  * **steps**: The same information found in the `predicate.buildConfig.steps` of a `taskrun` attestation.
+  * **invocation**: The same information found in the `predicate.invocation` of a `taskrun` attestation. This is needed to ensure default parameters on the task are visible in the attestation. It also maps pipeline parameters to task parameters.
+  * **results**: An array of objects representing the results of a `taskrun`. Each object contains the attribute `name` and `value`. Their values are of type string. `taskrun` results are important in the context of a `pipelinerun` because `taskrun` results may be used by other tasks, thus impacting the output of the `pipelinerun`.
+* **predicate.materials**: For `taskruns`, Chains populates this object based on specially named task parameters or results, i.e. `CHAINS-GIT_COMMIT` and `CHAINS-GIT_URL`. Similarly, Chains must populate this object with the corresponding pipeline parameters or results.
```

Comment:
`CHAINS-GIT_COMMIT` and `CHAINS-GIT_URL` are used today by Chains to populate the `predicate.materials` of taskrun attestations. I don't like them either! :smile: 

I think [TEP-109](https://github.com/tektoncd/community/pull/697) is a promising way to solve this.

For TEP-84, it would be beneficial to carry the existing behavior to pipelinerun attestations for the sake of consistency at least. WDYT?